File: spring/high-volume-traffic/backend/backend/src/main/java/com/jisu/backend/controller/CommentController.java
package com.jisu.backend.controller;

import com.jisu.backend.dto.WriteCommentDto;
import com.jisu.backend.entity.Comment;
import com.jisu.backend.service.CommentService;
import lombok.RequiredArgsConstructor;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.DeleteMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.PutMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/api/boards/{boardId}/articles/{articleId}/comments")
@RequiredArgsConstructor
public class CommentController {

  private final CommentService commentService;

  @PostMapping
  public ResponseEntity<Comment> writeComment(@PathVariable Long boardId,
      @PathVariable Long articleId,
      @RequestBody WriteCommentDto writeCommentDto) {
    return ResponseEntity.ok(commentService.writeComment(boardId, articleId, writeCommentDto));
  }

  @PutMapping("/{commentId}")
  public ResponseEntity<Comment> editComment(@PathVariable Long boardId,
      @PathVariable Long articleId,
      @PathVariable Long commentId,
      @RequestBody WriteCommentDto editCommentDto
  ) {
    return ResponseEntity.ok(
        commentService.editComment(boardId, articleId, commentId, editCommentDto));
  }

  @DeleteMapping("/{commentId}")
  public ResponseEntity<Comment> deleteComment(@PathVariable Long boardId,
      @PathVariable Long articleId,
      @PathVariable Long commentId
  ) {
    return ResponseEntity.ok(commentService.deleteComment(boardId, articleId, commentId));
  }

}

File: kotlin/kotlin-issue-service/issue-service/src/main/kotlin/com/example/kotlinissueservice/web/CommentController.kt
package com.example.kotlinissueservice.web

import com.example.kotlinissueservice.config.AuthUser
import com.example.kotlinissueservice.model.CommentRequest
import com.example.kotlinissueservice.model.CommentResponse
import com.example.kotlinissueservice.service.CommentService
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/v1/issues/{issueId}/comments")
class CommentController(
        private val commentService: CommentService,
) {

    @PostMapping
    fun create(
            authUser: AuthUser,
            @PathVariable issueId: Long,
            @RequestBody request: CommentRequest,
    ): CommentResponse {
        return commentService.create(issueId, authUser.userId, authUser.username, request)
    }

    @PutMapping("/{id}")
    fun edit(
            authUser: AuthUser,
            @PathVariable issueId: Long,
            @RequestBody request: CommentRequest,
    ) = commentService.edit(issueId, authUser.userId, request)

    @DeleteMapping("/{id}")
    @ResponseStatus(HttpStatus.NO_CONTENT)
    fun delete(
            authUser: AuthUser,
            @PathVariable issueId: Long,
            @PathVariable id: Long,
    ) {
        commentService.delete(issueId, id, authUser.userId)
    }
}